### PR TITLE
Password Generation API

### PR DESCRIPTION
**Pull request by master-fury**
### Implementations
**Updated Files**
- _/pom.xml_
    * Added passay library for password generation.

**New Files**
- _/api/GeneratePassword_
    * Created GeneratePassword API to allow users to generate random passwords.
    * Receives the data via GET request i.e. length passes to /actions/GeneratePasswordOperationsImpl
    * Returns GeneratePasswordOutput as response.
    * Handled all errors and exceptions
- _/model/GeneratePasswordOutput_
    * Model for sending status and password as response
- _/actions/GeneratePasswordOperations_
    * It is interface to handle password generation function.
- _/actions/GeneratePasswordOperationsImpl_
    * It is implementation class of GeneratePasswordOperations. 

**Functions**
* _/actions/GeneratePasswordOperationsImpl-generatePassword()_
    * Created generatePassword function to generate a random password that uses numbers, symbols, and English alphabets with the help of Passay Library.

**Other**
* Added logs in all above functions

**Testing**
* Did testing using postman by sending a request and receiving a response.

### CODE REVIEW

**Comment by master-fury** [comment]
sorry, It was some error while pushing, check now.

File: src/main/java/com/psipher/application/api/GeneratePassword.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    @RequestMapping(path = "/generatepassword", method = RequestMethod.GET)
+    public GeneratePasswordOutput generatePassword(@RequestParam Integer length) {
+        GeneratePasswordOutput generatePasswordOutput = new GeneratePasswordOutput();
+        try {
+            if (length != null && length >= 6) {
+                generatePasswordOutput.setPassword(generatePasswordOperations.generatePassword(length));
+                generatePasswordOutput.setStatus("success");
+            }
+        } catch (DDBException e) {
```
Comment (rahulkrishnan221) [inline src/main/java/com/psipher/application/api/GeneratePassword.java:33]: Why do we need try catch here? We are not using ddb for password generation then why do we need a ddb exception

File: src/main/java/com/psipher/application/api/GeneratePassword.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    @Autowired
+    public GeneratePassword(GeneratePasswordOperations generatePasswordOperations) {
+        this.generatePasswordOperations = generatePasswordOperations;
+    }
+
+    @RequestMapping(path = "/generatepassword", method = RequestMethod.GET)
+    public GeneratePasswordOutput generatePassword(@RequestParam Integer length) {
+        GeneratePasswordOutput generatePasswordOutput = new GeneratePasswordOutput();
+        try {
+            if (length != null && length >= 6) {
```
Comment (rahulkrishnan221) [inline src/main/java/com/psipher/application/api/GeneratePassword.java:29]: Add condition for else part

File: src/main/java/com/psipher/application/actions/GeneratePasswordOperations.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+ * Functions to generate random passwords.
+ */
+
+public interface GeneratePasswordOperations {
+    /**
+     * It generates a random password for users.
+     * @param length Length of password
+     * @return password
+     */
+    String generatePassword(Integer length) throws DDBException;
```
Comment (rahulkrishnan221) [inline src/main/java/com/psipher/application/actions/GeneratePasswordOperations.java:15]: You dont need to throw this exception it is not relevant

File: src/main/java/com/psipher/application/api/GeneratePassword.java (hunk 1 above)
Comment (master-fury) [reply src/main/java/com/psipher/application/api/GeneratePassword.java]: I got your point, I have made changes. Please check.

File: src/main/java/com/psipher/application/api/GeneratePassword.java (hunk 2 above)
Comment (master-fury) [reply src/main/java/com/psipher/application/api/GeneratePassword.java]: Added else part.

File: src/main/java/com/psipher/application/actions/GeneratePasswordOperations.java (hunk 3 above)
Comment (master-fury) [reply src/main/java/com/psipher/application/actions/GeneratePasswordOperations.java]: I got your point, I have made changes. Please check.

File: src/main/java/com/psipher/application/api/GeneratePassword.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    @Autowired
+    public GeneratePassword(GeneratePasswordOperations generatePasswordOperations) {
+        this.generatePasswordOperations = generatePasswordOperations;
+    }
+
+    @RequestMapping(path = "/generatepassword", method = RequestMethod.GET)
+    public GeneratePasswordOutput generatePassword(@RequestParam Integer length) {
+        GeneratePasswordOutput generatePasswordOutput = new GeneratePasswordOutput();
+        if (length != null && length >= 6) {
```
Comment (rahulkrishnan221) [inline src/main/java/com/psipher/application/api/GeneratePassword.java:23]: Why are we allowing at 6 why cant me >=8 ? Let me know your views on it

File: src/main/java/com/psipher/application/actions/GeneratePasswordOperationsImpl.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+            @Override
+            public String getErrorCode() {
+                return "Failed to generate Special Characters";
+            }
+
+            @Override
+            public String getCharacters() {
+                return "!\\\"#$%&'()*+,-./:;<=>?@[\\\\]^_`{|}~";
+            }
```
Comment (rahulkrishnan221) [inline src/main/java/com/psipher/application/actions/GeneratePasswordOperationsImpl.java:34]: move to static variable for constants

File: src/main/java/com/psipher/application/api/GeneratePassword.java (hunk 4 above)
Comment (master-fury) [reply src/main/java/com/psipher/application/api/GeneratePassword.java:23]: The usual minimum length for a password is 6, that's why I have selected it.

File: src/main/java/com/psipher/application/actions/GeneratePasswordOperationsImpl.java (hunk 5 above)
Comment (master-fury) [reply src/main/java/com/psipher/application/actions/GeneratePasswordOperationsImpl.java:34]: done, changes are made

**Pull request merged**